sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java

```diff
@@ -18,126 +18,95 @@
+import static org.apache.beam.sdk.io.elasticsearch.ElasticsearchIO.Read;
+import static org.apache.beam.sdk.io.elasticsearch.ElasticsearchIO.Write;
+import static org.apache.beam.sdk.io.elasticsearch.ElasticsearchIO.parseResponse;
 import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.core.Is.isA;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThat;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import java.io.IOException;
 import java.io.Serializable;
-import java.net.ServerSocket;
+import java.util.Collections;
 import java.util.List;
-import org.apache.beam.sdk.io.BoundedSource;
 import org.apache.beam.sdk.options.PipelineOptions;
 import org.apache.beam.sdk.options.PipelineOptionsFactory;
 import org.apache.beam.sdk.testing.PAssert;
-import org.apache.beam.sdk.testing.SourceTestUtils;
 import org.apache.beam.sdk.testing.TestPipeline;
 import org.apache.beam.sdk.transforms.Count;
 import org.apache.beam.sdk.transforms.Create;
 import org.apache.beam.sdk.transforms.DoFnTester;
 import org.apache.beam.sdk.values.PCollection;
-import org.elasticsearch.action.search.SearchResponse;
+import org.apache.http.HttpEntity;
+import org.apache.http.entity.ContentType;
+import org.apache.http.nio.entity.NStringEntity;
+import org.elasticsearch.client.Response;
 import org.elasticsearch.client.RestClient;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.index.query.QueryBuilder;
-import org.elasticsearch.index.query.QueryBuilders;
-import org.elasticsearch.node.Node;
 import org.hamcrest.CustomMatcher;
-import org.junit.AfterClass;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
-import org.junit.Rule;
-import org.junit.Test;
 import org.junit.rules.ExpectedException;
-import org.junit.rules.TemporaryFolder;
-import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Tests for {@link ElasticsearchIO}. */
-@RunWith(JUnit4.class)
-public class ElasticsearchIOTest implements Serializable {
+/** Common test class for {@link ElasticsearchIO}. */
+class ElasticsearchIOTestCommon implements Serializable {
 
-  private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchIOTest.class);
+  private static final Logger LOG = LoggerFactory.getLogger(ElasticsearchIOTestCommon.class);
 
-  private static final String ES_INDEX = "beam";
-  private static final String ES_TYPE = "test";
-  private static final String ES_IP = "127.0.0.1";
-  private static final long NUM_DOCS = 400L;
   private static final int NUM_SCIENTISTS = 10;
   private static final long BATCH_SIZE = 200L;
-  private static final long AVERAGE_DOC_SIZE = 25L;
   private static final long BATCH_SIZE_BYTES = 2048L;
 
-  private static Node node;
-  private static RestClient restClient;
-  private static ElasticsearchIO.ConnectionConfiguration connectionConfiguration;
+  private long numDocs;
```

Comment:
Not sure these need to be instance variables - I think it could be easier to understand if they were just passed to individual static methods of this class.

Reply:
The aim was to have a coherent behavior of `ElasticsearchIOTestCommon`: either it is used as UTests and `numDocs==400L` and we need to insert tests data, or it is used as ITests (high volume test on external ES) and `numDocs==50000L`  and we do not insert test data because they are supposed to be already in ES (See `ElasticsearchIOITCommon.main()`)